src/main/java/frc/robot/subsystems/algaeintake/AlgaeIntakeSubsystem.java

```diff
@@ -0,0 +1,154 @@
+  private final PositionVoltage positionRequest = new PositionVoltage(0).withSlot(0);
+
+  public AlgaeIntakeSubsystem() {
+    this.clawMotor = new TalonFX(AlgaeIntakeConstants.kClawMotorId);
+    this.absEncoder = new CANcoder(AlgaeIntakeConstants.kAbsoluteEncoderChannel);
+    this.beamBreak = new DigitalInput(AlgaeIntakeConstants.kBeambreakid);
+    clawMotor.getConfigurator().apply(AlgaeIntakeConstants.kMotionMagicConfig);
+  }
+
+  private final CombinedAlert angleAlert =
+      new CombinedAlert(
+          CombinedAlert.Severity.ERROR,
+          "Wrist Angle Error",
+          "The Algae Intake angle is outside the safe range. Subsystem disabled.");
+  private final CombinedAlert velocityAlert =
+      new CombinedAlert(
+          CombinedAlert.Severity.ERROR,
+          "Wrist Velocity Error",
+          "The Algae Intake velocity is outside the safe range. Subsystem disabled.");
+
+  private final CombinedAlert overheatingAlert =
+      new CombinedAlert(
+          CombinedAlert.Severity.ERROR,
+          "Wrist Overheating",
+          "The Algae Intake motor is overheating. Subsystem disabled.");
+
+  private enum ControlMode {
+    MANUAL,
+    MOTIONMAGIC
+  }
+
+  private ControlMode controlMode = ControlMode.MANUAL;
+
+  public void setPower(double power) {
+    if (controlMode == ControlMode.MOTIONMAGIC || !safetyCheck()) return;
+    power = Math.max(-1, Math.min(1, power));
+    clawMotor.setVoltage(power * 12);
+  }
+
+  public Boolean getBeamBreak() {
+    return beamBreak.get();
+  }
+
+  private void toPosition(Angle angle) {
+    controlMode = ControlMode.MOTIONMAGIC;
+    if (!safetyCheck()) return;
+    clawMotor.setControl(positionRequest.withPosition(angle));
+  }
+
+  private void switchControlMode(ControlMode control) {
+    controlMode = control;
+  }
+
+  public Command setPosition(Angle angle) {
+    return this.runEnd(() -> toPosition(angle), () -> switchControlMode(ControlMode.MANUAL));
+  }
+
+  public void setSpeed(AngularVelocity speed) {
+    clawMotor.setControl(velocityRequest.withVelocity(speed));
+  }
+
+  public void coast() {
+    clawMotor.setNeutralMode(NeutralModeValue.Coast);
+  }
+
+  public void setClosingVoltage(double voltage) {
+    clawMotor.setVoltage(voltage);
+  }
+
+  public void setOpeningVoltage(double voltage) {
+    clawMotor.setVoltage(-voltage);
+  }
+
+  public void brake() {
+    clawMotor.setNeutralMode(NeutralModeValue.Brake);
+  }
+
+  public Angle getPosition() {
+    return absEncoder.getPosition().getValue();
+  }
+
+  public double getPositionDouble() {
+    return clawMotor.getPosition().getValueAsDouble();
+  }
+
+  public AngularVelocity getVelocity() {
+    return clawMotor.getVelocity().getValue();
+  }
+
+  public double getVelocityAsDouble() {
+    return clawMotor.getVelocity().getValueAsDouble();
+  }
+
+  @Override
+  public void periodic() {
+    SmartDashboard.putNumber("Claw Velocity", absEncoder.getVelocity().getValueAsDouble());
```

Comment:
Should use a unit, never getValueAsDouble.
```suggestion
    SmartDashboard.putNumber("Claw Velocity", absEncoder.getVelocity().in(MetersPerSecond));
```

Reply:
apparently those methods arent used anywhere i shouldve deleted them earlier lmaoooooo